### PR TITLE
[curweather] x() no longer used and more

### PR DESCRIPTION
**Pull request by Quix0r**
Fixes/rewrites:
- install/uninstall functions moved up (common across addons)
- Added `App` as type-hint for $a and no & (call-by-reference) anymore
- use empty() instead of deprecated x()
- opening curly brace belongs into new line after function/method declaration
- added a lot other curly braces for better readability
- added spaces and empty lines

### CODE REVIEW

File: curweather/curweather.php [hunk 1]
```diff
@@ -12,84 +12,96 @@
...
+{
+	Addon::unregisterHook('network_mod_init'   , 'addon/curweather/curweather.php', 'curweather_network_mod_init');
+	Addon::unregisterHook('addon_settings'     , 'addon/curweather/curweather.php', 'curweather_addon_settings');
+	Addon::unregisterHook('addon_settings_post', 'addon/curweather/curweather.php', 'curweather_addon_settings_post');
+}
+
 //  get the weather data from OpenWeatherMap
-function getWeather( $loc, $units='metric', $lang='en', $appid='', $cachetime=0) {
-	$url = "http://api.openweathermap.org/data/2.5/weather?q=".$loc."&appid=".$appid."&lang=".$lang."&units=".$units."&mode=xml";
+function getWeather($loc, $units = 'metric', $lang = 'en', $appid = '', $cachetime = 0) {
```
Comment (MrPetovan) [inline curweather/curweather.php]: Can you please add a line feed before the brace while you're at it?

File: curweather/curweather.php [hunk 2]
```diff
@@ -12,84 +12,96 @@
...
 			$_SESSION['curweather_notice_shown'] = true;
 		}
+
 		return false;
 	}
+
 	unset($_SESSION['curweather_notice_shown']);
-	if ((string)$res->temperature['unit']==='metric') {
+
+	if ((string) $res->temperature['unit']==='metric') {
```
Comment (MrPetovan) [inline curweather/curweather.php]: Please add spaces around operators.

File: curweather/curweather.php [hunk 3]
```diff
@@ -12,84 +12,96 @@
...
 		'temperature' => (string) $res->temperature['value'][0].$tunit,
-		'pressure' => (string) $res->pressure['value'].(string)$res->pressure['unit'],
-		'humidity' => (string) $res->humidity['value'].(string)$res->humidity['unit'],
+		'pressure' => (string) $res->pressure['value'] . (string) $res->pressure['unit'],
+		'humidity' => (string) $res->humidity['value'] . (string) $res->humidity['unit'],
 		'descripion' => $desc,
-		'wind' => (string)$res->wind->speed['name'].' ('.(string)$res->wind->speed['value'].$wunit.')',
-		'update' => (string)$res->lastupdate['value'],
-		'icon' => (string)$res->weather['icon']
+		'wind' => (string) $res->wind->speed['name'] . ' (' . (string) $res->wind->speed['value'].$wunit.')',
```
Comment (MrPetovan) [inline curweather/curweather.php]: Missing spaces around concatenation operator.

File: curweather/curweather.php [hunk 4]
```diff
@@ -12,84 +12,96 @@
...
 		$tunit = '°C';
 		$wunit = 'm/s';
 	} else {
 		$tunit = '°F';
 		$wunit = 'mph';
 	}
-	if ( trim((string)$res->weather['value']) == trim((string)$res->clouds['name']) ) {
-		$desc = (string)$res->clouds['name'];
+
+	if ( trim((string) $res->weather['value']) == trim((string) $res->clouds['name']) ) {
```
Comment (MrPetovan) [inline curweather/curweather.php]: Please remove spaces inside parentheses.

File: curweather/curweather.php [hunk 5]
```diff
@@ -12,84 +12,96 @@
...
-function curweather_uninstall() {
-	Addon::unregisterHook('network_mod_init', 'addon/curweather/curweather.php', 'curweather_network_mod_init');
-	Addon::unregisterHook('addon_settings', 'addon/curweather/curweather.php', 'curweather_addon_settings');
-	Addon::unregisterHook('addon_settings_post', 'addon/curweather/curweather.php', 'curweather_addon_settings_post');
-}
-
-function curweather_network_mod_init(&$fk_app,&$b) {
-
-	if(! intval(PConfig::get(local_user(),'curweather','curweather_enable')))
+	if (! intval(PConfig::get(local_user(),'curweather','curweather_enable'))) {
```
Comment (MrPetovan) [inline curweather/curweather.php]: Please add a space after commas.

File: curweather/curweather.php [hunk 6]
```diff
@@ -183,28 +206,38 @@ function curweather_addon_settings(&$a,&$s) {
...
 	]);
+
 	return;
 }
 
 // Config stuff for the admin panel to let the admin of the node set a APPID
 // for accessing the API of openweathermap
-function curweather_addon_admin_post (&$a) {
-	if(! is_site_admin())
+function curweather_addon_admin_post (App $a)
```
Comment (MrPetovan) [inline curweather/curweather.php]: Please remove space before argument list parenthesis.

File: curweather/curweather.php [hunk 7]
```diff
@@ -12,84 +12,97 @@
...
 		$tunit = '°C';
 		$wunit = 'm/s';
 	} else {
 		$tunit = '°F';
 		$wunit = 'mph';
 	}
-	if ( trim((string)$res->weather['value']) == trim((string)$res->clouds['name']) ) {
-		$desc = (string)$res->clouds['name'];
+
+	if (trim((string) $res->weather['value']) == trim((string) $res->clouds['name']) ) {
```
Comment (MrPetovan) [inline curweather/curweather.php]: There still is an extra space at the end of the condition.

File: curweather/curweather.php [hunk 8]
```diff
@@ -12,84 +12,97 @@
...
-function curweather_uninstall() {
-	Addon::unregisterHook('network_mod_init', 'addon/curweather/curweather.php', 'curweather_network_mod_init');
-	Addon::unregisterHook('addon_settings', 'addon/curweather/curweather.php', 'curweather_addon_settings');
-	Addon::unregisterHook('addon_settings_post', 'addon/curweather/curweather.php', 'curweather_addon_settings_post');
-}
-
-function curweather_network_mod_init(&$fk_app,&$b) {
-
-	if(! intval(PConfig::get(local_user(),'curweather','curweather_enable')))
+	if (! intval(PConfig::get(local_user(), 'curweather','curweather_enable'))) {
```
Comment (MrPetovan) [inline curweather/curweather.php]: Please remove the space after the ! operator.

File: curweather/curweather.php [hunk 9]
```diff
@@ -183,28 +207,38 @@ function curweather_addon_settings(&$a,&$s) {
...
 	return;
 }
 
 // Config stuff for the admin panel to let the admin of the node set a APPID
 // for accessing the API of openweathermap
-function curweather_addon_admin_post (&$a) {
-	if(! is_site_admin())
+function curweather_addon_admin_post(App $a)
+{
+	if (! is_site_admin()) {
```
Comment (MrPetovan) [inline curweather/curweather.php]: Please remove the space after the ! operator.

Comment (Quix0r) [reply curweather/curweather.php]: All done. And I found a few missing spaces more.

File: curweather/curweather.php (hunk 7 above)
Comment (MrPetovan) [reply curweather/curweather.php]: This is still valid after the latest commit.

Comment (Quix0r) [reply curweather/curweather.php]: Now finally found. :-)

**Comment by Quix0r** [comment]
Maybe remove (strong) cast?

**Pull request merged**